### PR TITLE
Fixed typos in documentation

### PR DESCRIPTION
**Pull request by edoffagne**
## Description

Fixed typos in the following files:
- newbs_flashing.md
- arm_debugging.md
- cli.md
- hardware_keyboard_guidelines.md

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [X] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/arm_debugging.md [hunk 1]
```diff
@@ -26,10 +26,10 @@ If you are using windows you need to install this!
 
 `xpm install --global @gnu-mcu-eclipse/windows-build-tools`
 
-### Programer/Debugger Drivers
+### Programmer/Debugger Drivers
 
-Now its the time to install your programer's drivers. This tutorial was made using an ST-Link v2 which you can get from almost anywhere.
-If you have an ST-Link the drivers can be found [here](https://www.st.com/en/development-tools/stsw-link009.html) otherwise consult the manufuturer of your tool.
+Now its the time to install your programmer's drivers. This tutorial was made using an ST-Link v2 which you can get from almost anywhere.
```
Comment (fauxpark) [inline docs/arm_debugging.md]:
```suggestion
Now it's time to install your programmer's drivers. This tutorial was made using an ST-Link v2 which you can get from almost anywhere.
```

File: docs/arm_debugging.md [hunk 2]
```diff
@@ -6,15 +6,15 @@ This guide is catered towards advance users and assumes you can compile an ARM c
...
-The main objective here is to get the MCU Eclipse IDE correcly installed on our machine. The necesarry instructions are derived from [this](https://gnu-mcu-eclipse.github.io/install/) install guide.
+The main objective here is to get the MCU Eclipse IDE correctly installed on our machine. The necessary instructions are derived from [this](https://gnu-mcu-eclipse.github.io/install/) install guide.
 
 ### The xPack Manager
 
-This tool is a software package manager and it is used to help us get the necesarry depencencies.
+This tool is a software package manager and it is used to help us get the necessary dependencies.
 
-XPM runs using Node.js so grab that form [here](https://nodejs.org/en/). After installation, open a terminal and type `npm -v`. A reply with the version number means that the instalation was successful.
+XPM runs using Node.js so grab that form [here](https://nodejs.org/en/). After installation, open a terminal and type `npm -v`. A reply with the version number means that the installation was successful.
```
Comment (fauxpark) [inline docs/arm_debugging.md]:
```suggestion
XPM runs using Node.js so grab that from [here](https://nodejs.org/en/). After installation, open a terminal and type `npm -v`. A reply with the version number means that the installation was successful.
```

**Comment by kichloo** [comment]
I have fixed 2 issues.

**Comment by raghavdhingra** [comment]
Looks good. I had gone through the documentation.

**Pull request merged**